### PR TITLE
docs(guide): add .client and .server components documentation

### PR DESCRIPTION
**Pull request by huang-julien**
### 🔗 Linked issue
Hi 👋 I couldn't find any documentation about the .client components even if there was some *.client.vue components in the examples . This PR adds a small documentation for *.client.vue components, it also warn that directly importing a component (with direct path) does not work.

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Review by danielroe — APPROVED** [review]
Thank you! ❤️

This looks good to me for now, but when https://github.com/nuxt/framework/pull/5689 is merged we will need to refactor this.

File: docs/content/2.guide/3.directory-structure/4.components.md [hunk 1]
```diff
@@ -182,6 +182,47 @@ Use a slot as fallback until `<ClientOnly>` is mounted on client side.
...
+<template>
+  <div>
+    <!-- this component will only be rendered on client side -->
+    <Comments />
+  </div>
+</template>
+```
+
+::alert{type=warning}
+This feature only works with Nuxt auto-imports. Explicitly importing theses components does not convert them into client-only components.
```
Comment (danielroe) [inline docs/content/2.guide/3.directory-structure/4.components.md:204]:
```suggestion
This feature only works with Nuxt auto-imports. Explicitly importing these components does not convert them into client-only components.
```

**Comment by pi0** [comment]
Thanks

**Comment by Atinux** [comment]
Thank you ❤️

**Pull request merged**